akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -168,6 +174,22 @@ import scala.util.control.NonFatal
   }
 }
 
+@InternalApi
+private[typed] final class OptimizedActorAdapter[T](_initialBehavior: Behavior[T]) extends ActorAdapter[T](_initialBehavior) {
```

Comment:
can `ActorAdapter` become `abstract class` now?

Reply:
Ah, yes it should be

Reply:
perhaps we could let `receive` be abstract in `ActorAdapter` and remove `running` from that one and let  GuardianActorAdapter implement the become as it like?

Reply:
Yes, maybe that would be cleaner.